### ISSUE TITLE
Layers view shown in wrong place

### ISSUE THREAD

**Issue by nophead** [daid/LegacyCura#518]
I have a machine with the origin in the middle of the bed. I ticked this option in the startup wizard and the gcode seems correctly placed but the layers view of an object in the centre of the bed appears in the front left corner. This is on 13.06.4

**Comment by animacide** [comment]
I was able to fix this by making a change to sceneView.py on line 888, replace

```
            glTranslate(-self._machineSize[0] / 2, -self._machineSize[1] / 2, 0)
```

With:

```
            if profile.getPreference('machine_center_is_zero') != 'True': 
                glTranslate(-self._machineSize[0] / 2, -self._machineSize[1] / 2, 0)
```

I have compiled on Debian and confirmed that this change fixes the issue.

I think the changes to the gcodeInterpreter.py done for #446 are not needed and may give an incorrect interpretation of existing gcode files if they are loaded on a copy of Cura configured for Delta printers.

**Comment by nophead** [comment]
Yes I agree #446 looks wrong. It seems to assume machines with the origin in the centre also home to the centre but that would be odd as the limit switches are at the end of the axes. My machine homes to (-100,-100,200). The fact #446 is wrong just means the first gcode move will be shown incorrectly. To be correct it would need more information about the machine. Maybe better to ignore G28 and not draw the first move after it.

**Comment by daid** [comment]
#446 is for delta bots which ARE at 0,0,0 in the center when they are homed. Are for the first move showing wrong, that's just cosmetic and I will most likely remove the first move after homing from the visual GCode because of different machines with different homing positions.

**Comment by nophead** [comment]
But how does #446 fix the issue reported, which seems to be the same as this issue and is fixed by @animacide's patch above? I don't see why a delta would home to 0,0,0 any more than any other machine. It is user preference where you home to. I am sure if I built a delta I would home to the top like I do on all my machines.

**Comment by daid** [comment]
Ok, yes, I really shouldn't care where you home. And yes, the patch that I did is wrong. What I mend to say, deltabots have 0,0,0 at the center. Where they home only causes a different visual move the first GCode line, which is really just cosmetic.

**Comment by animacide** [comment]
Not sure about other Delta printers, but mine (Rostock MAX running Polygonhell's branch of Repetier firmware), homes to 0,0,max_height because the limit switches are at the top of the towers.  It is just cosmetic as far as setting the center after homing when interpreting gcode, so I agree that the fix in #446 wouldn't be a real problem.  If fact, I don't think it would even be visible since 0,0,0 is the center of the bed on a Delta and if the lines start there you're really not going to see anything out of place.  Thanks Daid for adding the fix for the layer view!
